File: sol/2145/2145.py
class Solution:
    def numberOfArrays(self, differences: List[int], lower: int, upper: int) -> int:
        n = len(differences)
        orig = [0] * (n + 1)
        for i, diff in enumerate(differences):
            orig[i + 1] = orig[i] + diff
        return max(0, upper - lower - max(orig) + min(orig) + 1)

File: sol/2145/2145.cpp
struct Solution {
    int numberOfArrays(vector<int>& differences, int lower, int upper) {
        long maxi = 0, mini = 0, val = 0;
        for (int diff : differences){
            val += diff;
            maxi = max(maxi, val);
            mini = min(mini, val);
        }
        return max(0l, upper - lower - maxi + mini + 1);
    }
};
